### PR TITLE
Emoji validate

### PR DESCRIPTION
**Pull request by qurbat**

### CODE REVIEW

File: test_requirements.txt [hunk 1]
```diff
@@ -3,3 +3,4 @@ coverage
 coveralls
 psycopg2
 urllib3
+emoji
```
Comment (iambibhas) [inline test_requirements.txt]: This needs to go in the `requires` list inside `setup.py`. Not here.

File: baseframe/forms/validators.py [hunk 2]
```diff
@@ -319,6 +320,19 @@ def __call__(self, form, field):
...
+class IsEmoji(object):
+    """
+    Validate whether the supplied string is an emoji.
+    """
+    def __init__(self, message=None):
+        self.message = message
+
+    def __call__(self, form, field):
+        if field.data in emoji.UNICODE_EMOJI:
+            return
```
Comment (iambibhas) [inline baseframe/forms/validators.py]: No need for this `return`. Check for the invalid case and raise `ValidationError` if needed.

File: baseframe/forms/validators.py [hunk 3]
```diff
@@ -9,6 +9,7 @@
 from fractions import Fraction
 import datetime
 import re
+import emoji
```
Comment (iambibhas) [inline baseframe/forms/validators.py]: install `isort` and sort these imports. Or install `pre-commit` and let it do that.

File: baseframe/forms/validators.py [hunk 4]
```diff
@@ -319,6 +320,15 @@ def __call__(self, form, field):
             return
         else:
             raise ValidationError(self.message)
+            
+class IsEmoji(object):
+
```
Comment (jace) [inline baseframe/forms/validators.py]: Docstring required.

File: baseframe/forms/validators.py [hunk 5]
```diff
@@ -319,6 +320,15 @@ def __call__(self, form, field):
...
+    def __init__(self, message=None):
+        self.message = message
```
Comment (jace) [inline baseframe/forms/validators.py]: Fallback to a default message as in the other validators.

File: baseframe/forms/validators.py [hunk 6]
```diff
@@ -319,6 +320,15 @@ def __call__(self, form, field):
             return
         else:
             raise ValidationError(self.message)
+            
+class IsEmoji(object):
```
Comment (jace) [inline baseframe/forms/validators.py]: This validator was inserted between two validators that are related to each other. This one should have been before or after them. Move it.

**Comment by jace** [comment]
This is also missing tests, so it reduces overall test coverage.

**Pull request merged**